tests/chainer_tests/functions_tests/loss_tests/test_softmax_cross_entropy.py

```diff
@@ -13,36 +13,38 @@
 
 
 @testing.parameterize(*(testing.product({
-    'shape': [None, (2, 3), (2, 3, 2), (2, 3, 2, 2)],
-    'cache_score': [True, False],
-    'normalize': [True, False],
-    'ignore_index': [None, (slice(None),), (0,), (0, 1), (0, 1, 0)],
+    'shape_ignore': [(None, None),
+                     ((2, 3), (slice(None),)),
+                     ((2, 3, 2), (0,)),
+                     ((2, 3, 2, 2), (0, 1, 0))],
+    'cache_score': [True],
+    'normalize': [True],
+    'weight_apply': [True],
+    'enable_double_backprop': [True],
     'dtype': [numpy.float32],
-    'weight_apply': [False, True],
-    'enable_double_backprop': [False, True],
     'label_dtype': [numpy.int32],
 }) + testing.product({
-    'shape': [None, (2, 3), (2, 3, 2), (2, 3, 2, 2)],
+    'shape_ignore': [(None, None), ((2, 3, 2, 2), (0, 1, 0))],
     'cache_score': [False],
-    'normalize': [True],
-    'ignore_index': [(0, 1)],
-    'dtype': [numpy.float16, numpy.float32, numpy.float64],
```

Comment:
This fix removes test cases for float types except float32 and I feel it is a problem because a developer easily make a bug for other float dtypes when he/she writes a CUDA kernel which does not use generics.
Actually SCE is not one simple FunctionNode, but it dispatches process based on its argument. For example it makes completely different function node with `enable_double_backprop=True`  and `False` (see the source). So it has two function implementations. With and without `class_weight` it calls different kernels (it has two kernel codes in its implementation).
It is better to consider a policy to reduce the number of test cases.